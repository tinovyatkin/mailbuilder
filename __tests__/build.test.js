'use strict';

const { simpleParser } = require('mailparser');

const { simpleHtmlEmail } = require('../lib/build');

describe('email message building', () => {
  test('simpleHtmlEmail', async () => {
    const testMsg = {
      from: { name: 'John "Cool" Doe', address: 'john@doe.com' },
      to: [{ name: '–í–∞—Å—è –ü—É–ø–∫–∏–Ω', address: 'vasya@pupkin.com' }],
      bcc: [
        { name: 'One Spy', address: 'spy@kremlin.ru' },
        { name: 'Big Eye', address: 'fbi@whitehouse.org' },
      ],
      subject: 'Some üíÑ subject',
      html: `This is our <b>html</b> message content
      It's also <i>multiline</i> just in case`,
    };
    const res = simpleHtmlEmail(testMsg);
    expect(res).toMatchSnapshot();

    // testing with alternative full-blown library
    const parsed = await simpleParser(res);
    expect(parsed.headers.get('from').text).toBe(
      `${testMsg.from.name} <${testMsg.from.address}>`,
    );
    expect(parsed.headers.get('to').text).toBe(
      `${testMsg.to[0].name} <${testMsg.to[0].address}>`,
    );
    expect(parsed.subject).toBe(testMsg.subject);
    expect(parsed.bcc.value).toEqual(expect.arrayContaining(testMsg.bcc));
    expect(parsed.html).toBe(testMsg.html);
  });
});
