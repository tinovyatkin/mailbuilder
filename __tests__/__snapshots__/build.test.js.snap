// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`email message building simpleHtmlEmail 1`] = `
"Content-Type: text/html; charset=utf-8
Content-Transfer-Encoding: 8Bit
From: \\"John \\\\\\"Cool\\\\\\" Doe\\" <john@doe.com>
To: =?utf-8?B?0JLQsNGB0Y8g0J/Rg9C/0LrQuNC9?= <vasya@pupkin.com>
MIME-Version: 1.0
Bcc: One Spy <spy@kremlin.ru>, Big Eye <fbi@whitehouse.org>
Subject: =?utf-8?B?U29tZSDwn5KEIHN1YmplY3Q=?=

This is our <b>html</b> message content
      It's also <i>multiline</i> just in case"
`;
